### PR TITLE
Move NotifierService registration to IBootstrap

### PR DESCRIPTION
**Pull request by rullzer**
Yesterdays evening coding result

Todo:
- [x] Move the server apps over to use this if possible

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]
Love it

File: lib/private/Notification/Manager.php [hunk 1]
```diff
@@ -141,7 +147,7 @@ protected function getApps(): array {
 	 * @return INotifier[]
 	 */
 	public function getNotifiers(): array {
-		if (empty($this->notifierClasses)) {
+		if (!empty($this->notifiers)) {
```
Comment (nickvergessen) [inline lib/private/Notification/Manager.php:150]: this breaks on the fly registrations like in unit tests :-X

Comment (rullzer) [reply lib/private/Notification/Manager.php:150]: But then you keep adding the notifiers right? anyway let me do some other magic then

Comment (nickvergessen) [reply lib/private/Notification/Manager.php:150]: It clears the list with `$this->notifierClasses = [];` (old line 169 new line 197).
I think you just want to make sure your services code is only ran once and at the right time.

File: lib/private/Notification/Manager.php [hunk 2]
```diff
@@ -141,7 +150,33 @@ protected function getApps(): array {
...
+					continue;
+				}
+
+				$this->notifiers[] = $notifier;
+			}
+
+			$this->parsedRegistrationContext = true;
+		}
+
+		if (!empty($this->notifiers)) {
```
Comment (nickvergessen) [inline lib/private/Notification/Manager.php:179]:
```suggestion
		if (empty($this->notifierClasses)) {
```

And we are good to go

Comment (rullzer) [reply lib/private/Notification/Manager.php:179]: Yes. That change was still pending

File: apps/updatenotification/lib/AppInfo/Application.php [hunk 3]
```diff
@@ -67,9 +68,6 @@ public function boot(IBootContext $context): void {
 				return;
 			}
 
-			// Always register the notifier, so background jobs (without a user) can send push notifications
-			$notificationsManager->registerNotifierService(Notifier::class);
```
Comment (nickvergessen) [inline apps/updatenotification/lib/AppInfo/Application.php:71]: drop the injection of `$notificationsManager` as well

**Review by nickvergessen — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**